src/ext/io.rs

```diff
@@ -42,11 +45,38 @@ pub enum Error {
     #[error("IO layer error")]
     IO,
 
+    /// The destination directory for a file move operation could not be created.
+    #[error("create parent directory for destination {}", .0.display())]
+    CreateParentDir(PathBuf),
+
     /// Failed to join the background worker that performed the backing IO operation.
     #[error("join background worker")]
     JoinWorker,
 }
 
+/// Lists the contents of a directory.
+/// Returns the file names without their path components.
+#[tracing::instrument]
+pub async fn list_contents(dir: &Path) -> Result<Vec<String>, Report<Error>> {
```

Comment:
nit: I don't think this is used anywhere

Reply:
Yeah, but I wrote it for debugging and figure it'll be useful at some point.